lib/hammer_cli/subcommand.rb

```diff
@@ -44,13 +66,29 @@ def remove_subcommand(name)
         end
       end
 
-      def subcommand!(name, description, subcommand_class = self, &block)
+      def subcommand!(name, description, subcommand_class = self, options = {}, &block)
         remove_subcommand(name)
-        subcommand(name, description, subcommand_class, &block)
+        subcommand(name, description, subcommand_class, options, &block)
         logger.info "subcommand #{name} (#{subcommand_class}) was created."
       end
 
-      def subcommand(name, description, subcommand_class = self, &block)
+      def subcommand(name, description, subcommand_class = self, options = {}, &block)
+        definition = Definition.new(name, description, subcommand_class, options)
+        define_subcommand(subcommand_class, definition, &block)
+      end
+
+      def lazy_subcommand(name, description, subcommand_class_name, path, options = {})
+        definition = LazyDefinition.new(name, description, subcommand_class_name, path, options)
+        define_subcommand(Class, definition)
+      end
+
+      def lazy_subcommand!(name, description, subcommand_class_name, path, options = {})
+        remove_subcommand(name)
+        self.lazy_subcommand(name, description, subcommand_class_name, path, options)
+        logger.info "subcommand #{name} (#{subcommand_class_name}) was created."
+      end
+
+      def define_subcommand(subcommand_class, definition, &block)
         existing = find_subcommand(name)
```

Comment:
@rahulbajaj0509 ,

Does `name` accessible inside this method? Everytime you will get nil value for existing.

Reply:
Thanks @kgaikwad :)